File: test-suite/expose-tests/consumer-r/src.R
# .morloc.source is defined in the R pool preamble. With the expose
# mechanism, the path "morloc-test-expose-r/util.R" resolves via the
# .morloc.libdir fallback to $MORLOC_HOME/lib/R/morloc-test-expose-r/util.R.
.morloc.source("morloc-test-expose-r/util.R")

compute <- function(x) {
    doubler(util_add_one(x))
}

File: test-suite/expose-tests/consumer-py/src.py
# Imports the exposed Python subpackage. Resolution depends on
# $MORLOC_HOME/lib/python on sys.path (set by the py preamble) and on
# the install pipeline having mapped `morloc-test-expose-py` to the
# Python-legal name `morloc_test_expose_py`.
from morloc_test_expose_py.pkg.sub.util import util_add_one, doubler


def compute(x):
    return doubler(util_add_one(x))
